src/transport/grpc/mod.rs

```diff
@@ -622,13 +640,266 @@ impl<'a> TryFrom<&'a Interfaces> for NodeData {
+        if prop.ownership.is_device() {
+            // do not store device owned properties
+            return Ok(());
+        }
+
+        self.store
+            .store_prop(prop)
+            .await
+            .map_err(GrpcStoreError::from_err)
+    }
+
+    async fn load_prop<I>(
+        &self,
+        interface: &StoreInterfaceData<I>,
+        path: &str,
+        interface_major: i32,
+    ) -> Result<Option<AstarteType>, Self::Err>
+    where
+        I: AsRef<str> + Send + Sync,
+    {
+        if interface.ownership.is_server() {
+            return self
+                .store
+                .load_prop(interface, path, interface_major)
+                .await
+                .map_err(GrpcStoreError::from_err);
+        }
+
+        let property = self
+            .client
+            .lock()
+            .await
+            .get_property(PropertyIdentifier {
+                interface_name: interface.name.as_ref().to_owned(),
+                path: path.to_owned(),
+            })
+            .await
+            .map_err(GrpcStoreError::from_status)
+            .map(tonic::Response::into_inner)?;
+
+        convert::map_property_to_astarte_type(property).map_err(GrpcStoreError::from_conversion)
+    }
+
+    async fn unset_prop<I>(
+        &self,
+        interface: &StoreInterfaceData<I>,
+        path: &str,
+    ) -> Result<(), Self::Err>
+    where
+        I: AsRef<str> + Send + Sync,
+    {
+        if interface.ownership.is_device() {
+            // we won't store a device unset since we always request the properties to the server
+            return Ok(());
+        }
+
+        self.store
+            .unset_prop(interface, path)
+            .await
+            .map_err(GrpcStoreError::from_err)
+    }
+
+    async fn delete_prop<I>(
+        &self,
+        interface: &StoreInterfaceData<I>,
+        path: &str,
+    ) -> Result<(), Self::Err>
+    where
+        I: AsRef<str> + Send + Sync,
+    {
+        if interface.ownership.is_device() {
+            // we won't delete a device property since we always request the properties to the server
+            return Ok(());
+        }
+
+        self.store
+            .delete_prop(interface, path)
+            .await
+            .map_err(GrpcStoreError::from_err)
+    }
+
+    async fn clear(&self) -> Result<(), Self::Err> {
+        self.store.clear().await.map_err(GrpcStoreError::from_err)
+    }
+
+    async fn load_all_props(&self) -> Result<Vec<StoredProp>, Self::Err> {
+        let load_server = async {
+            self.store
+                .server_props()
+                .await
+                .map_err(GrpcStoreError::from_err)
+        };
+
+        let (server_result, device_result) =
+            futures::future::join(load_server, self.load_device_properties()).await;
+
+        let merged_properties = server_result?
+            .into_iter()
+            .chain(device_result?.into_iter())
+            .collect_vec();
```

Comment:
You could directly use the `append` method of vec (thus avoid importing `Itertools`:

```suggestion
let mut merged_properties = server_result?;
merged_properties.append(&mut device_result?);
```

Reply:
This avoids having a mut variable, itertools is already imported anyways, are you sure anyway about the change ?

Reply:
It would have sense using `chain` with iterators if you don't have to collect at the end. Also, append does not clone but rather move elements from one vec to the other

Reply:
In the end I applied the change